frontend/src/views/Gallery.vue

```diff
@@ -1,126 +1,209 @@
 <script setup>
-import axios from 'axios'
-import { ref, inject, onMounted } from 'vue'
-import { onBeforeRouteUpdate, useRoute } from 'vue-router'
+import axios from "axios";
+import { ref, inject, onMounted } from "vue";
+import { onBeforeRouteUpdate, useRoute } from "vue-router";
 import { io } from "socket.io-client";
-import { views } from '@/utils/utils.js'
-import { storeFilter } from '@/stores/filter.js'
-import { storeGalleryView } from '@/stores/galleryView.js'
-import { normalizeString } from '@/utils/utils.js'
-import FilterBar from '@/components/GameGallery/FilterBar.vue'
-import GalleryViewBtn from '@/components/GameGallery/GalleryViewBtn.vue'
-import GameCard from '@/components/GameGallery/Card/Base.vue'
-import GameListHeader from '@/components/GameGallery/ListItem/Header.vue'
-import GameListItem from '@/components/GameGallery/ListItem/Item.vue'
-import { storeScanning } from '@/stores/scanning.js'
-
+import { views } from "@/utils/utils.js";
+import { storeFilter } from "@/stores/filter.js";
+import { storeGalleryView } from "@/stores/galleryView.js";
+import { normalizeString } from "@/utils/utils.js";
+import FilterBar from "@/components/GameGallery/FilterBar.vue";
+import GalleryViewBtn from "@/components/GameGallery/GalleryViewBtn.vue";
+import GameCard from "@/components/GameGallery/Card/Base.vue";
+import GameListHeader from "@/components/GameGallery/ListItem/Header.vue";
+import GameListItem from "@/components/GameGallery/ListItem/Item.vue";
+import { storeScanning } from "@/stores/scanning.js";
+import { storeRenaming } from "@/stores/renaming.js";
 
 // Props
-const roms = ref([])
-const gettingRoms = ref(false)
-const filter = storeFilter()
-const galleryView = storeGalleryView()
-const route = useRoute()
-const romsFiltered = ref([])
+const roms = ref([]);
+const gettingRoms = ref(false);
+const filter = storeFilter();
+const galleryView = storeGalleryView();
+const route = useRoute();
+const romsFiltered = ref([]);
 // const firmwares = ["firmware_base", "firmware_bios"]
 // const sections = ['roms', 'firmwares']
-const currentSection = ref('roms')
-const scanning = storeScanning()
+const currentSection = ref("roms");
+const scanning = storeScanning();
+const renaming = storeRenaming();
 
 // Event listeners bus
-const emitter = inject('emitter')
-emitter.on('filter', () => { filterRoms() })
+const emitter = inject("emitter");
+emitter.on("filter", () => {
+  filterRoms();
+});
 
+const socket = io({
```

Comment:
Is it better to open a socket connection each time the view loads instead to open it just when scanning/renaming?

Reply:
Ideally we want a single connection for the entire app, which is something I'll try to do in this PR.

As for whether to keep that connection open all the time, or disconnect and reconnect it, it really depends on our use case. Opening and closing sockets is CPU intensive, and if the typical install won't be running more then a handful of connections (<100) at a time, then there's no hard in keeping them open. However, since rescan actions are probably not very common, there is a good argument for opening the socket just-in-time, and closing it after use.

Reply:
Oh okay, I think then we should go for a single app connection and keep it open then because I guess it will be useful in the future when RomM grows much more and as you mention this is not going to have hundreds of concurrent connections/users.